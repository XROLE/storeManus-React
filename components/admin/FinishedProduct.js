
import React, { Fragment, Component } from 'react';
import { connect } from 'react-redux';
import Spinner from 'react-spinkit';
import Navbar from '../navbar/Navbar';
import Sidebar from '../sidebar/Sidebar';
import TopBar from '../sidebar/TopBar';
import { getFinishedProducts } from '../../store/actions/actions';


class FinishedProducts extends Component {
  constructor(props) {
    super();
  }

  componentDidMount() {
    const { get, finishedProducts } = this.props;
    if (!finishedProducts) {
      get(getFinishedProducts());
    }
  }


  render() {
    const { finishedProducts, pending } = this.props;

    return (
      <Fragment>
        <Navbar />
        <div className="dasboardDiv">
          <Sidebar />
          <div className="dashboardDisplayDiv">
            <TopBar />
            <div className="dasboard-display-div fontBlack" style={{ width: '90%' }} id="allProductContainer">
              <table className="all-products-table" id="all-products-table">
                <thead>
                  <tr>
                    <th colSpan="5" className="table-head">
                      <i className="fas fa-cookie-bite" />
                      &nbsp;AVAILABLE PRODUCTS
                    </th>
                  </tr>
                </thead>
                <tbody id="tableBody">
                  <tr>
                    <th>ID</th>
                    <th>  Name</th>
                    <th>Categories</th>
                    <th>Quantity</th>
                    <th>Date Added</th>
                  </tr>
                  {finishedProducts && (
                    finishedProducts.map(product => (
                      <tr>
                        <td align="left">{product.id}</td>
                        <td align="left">{product.name}</td>
                        <td align="left">{product.category}</td>
                        <td align="left">{product.quantity}</td>
                        <td align="left">{product.date}</td>
                      </tr>
                    ))
                  )}
                </tbody>
              </table>
              { pending === true && (
                <Spinner name="circle" className="spinner" id="reactLoader" />
              )}
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}


const mapDispatchToProps = dispatch => ({
  get: value => dispatch(value),
});

const mapStateToProps = state => ({
  finishedProducts: state.finishedProducts.finishedProducts,
});

export default connect(mapStateToProps, mapDispatchToProps)(FinishedProducts);
