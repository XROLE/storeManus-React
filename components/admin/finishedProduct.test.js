import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import axios from 'axios';
import { BrowserRouter } from 'react-router-dom';
import FinishedProduct from './FinishedProduct';
import finishedProductReducer from '../../store/reducers/finisheProductReducer';

jest.mock('axios');

describe('Finished Products', () => {
  it('Should render just finished product component', () => {
    const component = shallow(
      <FinishedProduct />,
    );
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});

describe('Finished Products Component', () => {
  const get = jest.fn();
  const initialState = {
    finishedProducts: {
      message: '',
      pending: false,
      error: '',
      FinishedProduct: '',
    },
  };
  const mockStore = configureStore([thunk]);
  const store = mockStore(() => initialState);
  const dispatchMock = jest.spyOn(store, 'dispatch');
  const finishedProducts = [];

  const props = {
    finishedProducts: {
      finishedProducts,
      get,
    },
  };

  axios.get.mockImplementation(() => Promise.resolve({
    data: { finishedProducts: [{ name: 'Milo' }] },
  }));

  mount(
    <Provider store={store}>
      <BrowserRouter>
        <FinishedProduct {...props} finishedProducts />
      </BrowserRouter>
    </Provider>,
  );

  expect(dispatchMock).toHaveBeenCalled();
});

describe('Finished Products Component', () => {
  const get = jest.fn();
  const initialState = {
    finishedProducts: {
      message: '',
      pending: false,
      error: '',
      FinishedProduct: '',
    },
  };
  const mockStore = configureStore([thunk]);
  const store = mockStore(() => initialState);
  const dispatchMock = jest.spyOn(store, 'dispatch');
  const finishedProducts = [];

  const props = {
    finishedProducts: {
      finishedProducts,
      get,
    },
  };

  const error = {
    response: {
      data: {
        Message: 'An error occured',
      },
    },
  };
  axios.get.mockImplementation(() => Promise.reject(error));

  mount(
    <Provider store={store}>
      <BrowserRouter>
        <FinishedProduct {...props} finishedProducts />
      </BrowserRouter>
    </Provider>,
  );

  expect(dispatchMock).toHaveBeenCalled();
});

it('should setup default state values', () => {
  const state = finishedProductReducer(undefined, {
    type: '@@INIT',
  });
  expect(state).toEqual({
    message: '',
    pending: false,
    error: '',
    finishedProducts: '',
  });
});

it('should update state when action type is  FINISHED_PRODUCT_IN_PROGRESS', () => {
  const state = finishedProductReducer(undefined, {
    type: 'FINISHED_PRODUCT_IN_PROGRESS',
  });
  expect(state).toEqual({
    ...state,
    error: null,
    pending: true,
  });
});

it('should update state when action type is  FINISHED_PRODUCT_SUCCESS', () => {
  const state = finishedProductReducer(undefined, {
    type: 'FINISHED_PRODUCT_SUCCESS',
    payload: {
      finishedProducts: [{ tea: 'coffee' }],
    },
  });
  expect(state).toEqual({
    ...state,
    finishedProducts: [{ tea: 'coffee' }],
  });
});

it('should update state when action type is  FINISHED_PRODUCT_ERROR', () => {
  const state = finishedProductReducer(undefined, {
    type: 'FINISHED_PRODUCT_ERROR',
    error: 'failed',
  });
  expect(state).toEqual({
    ...state,
    error: 'failed',
    pending: false,
  });
});
