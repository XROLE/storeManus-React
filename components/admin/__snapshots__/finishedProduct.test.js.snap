// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Finished Products Should render just finished product component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
